modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeIndexBase.java

```diff
@@ -125,34 +82,76 @@ protected H2TreeIndexBase(GridH2Table tbl, String name, IndexColumn[] cols, Inde
      * @param maxInlineSize Max inline size from cache config.
      * @return Inline size.
      */
-    static int computeInlineSize(List<InlineIndexHelper> inlineIdxs, int cfgInlineSize, int maxInlineSize) {
-        int propSize = maxInlineSize == -1 ? IgniteSystemProperties.getInteger(IgniteSystemProperties.IGNITE_MAX_INDEX_PAYLOAD_SIZE,
-            IGNITE_MAX_INDEX_PAYLOAD_SIZE_DEFAULT) : maxInlineSize;
-
+    static int computeInlineSize(List<InlineIndexColumn> inlineIdxs, int cfgInlineSize, int maxInlineSize) {
         if (cfgInlineSize == 0)
             return 0;
 
         if (F.isEmpty(inlineIdxs))
             return 0;
 
-        if (cfgInlineSize == -1) {
-            if (propSize == 0)
-                return 0;
+        if (cfgInlineSize != -1)
+            return Math.min(PageIO.MAX_PAYLOAD_SIZE, cfgInlineSize);
 
-            int size = 0;
+        int propSize = maxInlineSize == -1 ? IgniteSystemProperties.getInteger(IgniteSystemProperties.IGNITE_MAX_INDEX_PAYLOAD_SIZE,
```

Comment:
getInteger() is synchronous HashTable operation.
Let's make is static.

Reply:
I suppose it could break a tests, which change the property. The method invoke so seldom, what shouldn't affect a performance

Reply:
agree with Yuri. I would prefer not to use static constant as it brings some troubles to testing.